lib/SPIRV/OCLTypeToSPIRV.cpp

```diff
@@ -250,8 +250,12 @@ void OCLTypeToSPIRV::adaptArgumentsBySamplerUse(Module &M) {
           AdaptedTy.count(SamplerArg) != 0) // Already traced this, move on.
         continue;
 
+      if (isSPIRVType(SamplerArg->getType(), kSPIRVTypeName::Sampler))
+        return;
```

Comment:
Maybe we should not return, but continue with rest of the users of the function, shouldn't we ?